src/utils/EventUtils.ts

```diff
@@ -0,0 +1,231 @@
+import { Result } from "@ethersproject/abi";
+import { delay } from "@uma/financial-templates-lib";
+import { Contract, Event, EventFilter } from "ethers";
+import { SortableEvent } from "../interfaces";
+
+const maxRetries = 3;
+const retrySleepTime = 10;
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function spreadEvent(args: Result | undefined): any {
+  if (!args) {
+    throw new Error("No args provided");
+  }
```

Comment:
Not sure this is a good idea - Event.args is generated by ethers and this is the first place that we inspect it. It's not required to be defined in Ethers, so I imagine there could be theoretical scenarios where we arrive here and throw an error on a legitimate Event (though not one emitted by our contracts).